barcode/__init__.py

```diff
@@ -1,7 +1,45 @@
 from uber.common import *
 from barcode._version import __version__
+from uber.reports import *
+from barcode import *
+from barcode.barcode_utils import generate_barcode_from_badge_num, get_badge_num_from_barcode
 
 config = parse_config(__file__)
 mount_site_sections(config['module_root'])
 static_overrides(join(config['module_root'], 'static'))
 template_overrides(join(config['module_root'], 'templates'))
+
+
+class BarcodeReportMixin:
+    """
+    Overridden methods from the Reports base class.
+    We will inject a barcode row into the exported data
+    """
+
+    def write_row(self, row, out):
+        if self._include_badge_nums:
+            # add in the barcodes here
+            badge_num = row[0]
+            barcode = generate_barcode_from_badge_num(badge_num)
+            row.append(barcode)
+
+        # TODO: we should call the base class's writeRow(), but that's complex because of our multiple inheritance.
+        # this works fine for now.
+        out.writerow(row)
+
+
+class BarcodePrintedBadgeReport(BarcodeReportMixin, uber.reports.PrintedBadgeReport):
+    pass
+
+
+class BarcodePersonalizedBadgeReport(BarcodeReportMixin, uber.reports.PersonalizedBadgeReport):
+    pass
+
+
+class BarcodeSupporterBadgeReport(BarcodeReportMixin, uber.reports.SupporterBadgeReport):
+    pass
+
+
+uber.reports.printed_badge_report_type = BarcodePrintedBadgeReport
+uber.reports.personalized_badge_report_type = BarcodePersonalizedBadgeReport
+uber.reports.supporter_badge_report_type = BarcodeSupporterBadgeReport
```

Comment:
Actually, I believe the way Sideboard inherits plugins lets you straight-up write functions that are called the same thing and have them overridden automatically.

Python doesn't have a problem with two functions with the same name - it just uses the latest one it finds. So assuming that your plugin's functions are read after core's, your functions will automatically be used in place of them.

We can test my theory later, though. :)